C4.puml

```diff
@@ -141,30 +191,74 @@ $tagSkin
   !endif
   !$tagEntry = $tagEntry + " " + $tagStereo + " "
   !if ($shadowing == "true")
-    !$tagEntry = $tagEntry + "(shadow) "
+    !$tagEntry = $tagEntry + $LEGEND_SHADOW_TEXT
   !endif
   !if ($shadowing == "false")
-    !$tagEntry = $tagEntry + "(no shadow) "
+    !$tagEntry = $tagEntry + $LEGEND_NO_SHADOW_TEXT
   !endif
   !if ($fontColor == "" && $bgColor == "")
-      !$tagEntry = $tagEntry + "(no font, no back color) "
+      !$tagEntry = $tagEntry + $LEGEND_NO_FONT_BG_TEXT
   !else 
     !if ($fontColor == "")
-      !$tagEntry = $tagEntry + "(no font color) "
+      !$tagEntry = $tagEntry + $LEGEND_NO_FONT_TEXT
     !endif
     !if ($bgColor == "")
-      !$tagEntry = $tagEntry + "(no back color) "
+      !$tagEntry = $tagEntry + $LEGEND_NO_BG_TEXT
     !endif
   !endif
   !$tagEntry = $tagEntry + "</color> "
   !$tagEntry = $tagEntry + "|"
   !return $tagEntry 
 !endfunction
 
-!unquoted procedure $addTagToDynamicLegend($tagStereo, $bgColor="", $fontColor="", $borderColor="", $shadowing="")
-  !$tagEntry = $tagLegendEntry($tagStereo, $bgColor, $fontColor, $borderColor, $shadowing)
+!function $tagRelLegendEntry($tagStereo, $textColor, $lineColor)
+  !$tagEntry = "|"
+  !$bg = $LEGEND_UNDEFINED_BG_COLOR
+  ' named colors have to start with # too 
+  !if (%substr($bg, 0, 1) != "#")
+    !$bg = "#" + $bg
+  !endif
+'  !$tagEntry = $tagEntry + "<" + $bg +">" 
```

Comment:
Here and further down, there is commented-out code. If it is meant as documentation, its meaning is unclear to me. If it is not meant as documentation, it might just need to be removed.

Reply:
cleaned source and updated some comments